stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/call/RtcSession.kt

```diff
@@ -879,59 +880,36 @@ public class RtcSession internal constructor(
      * Change the quality of video we upload when the ChangePublishQualityEvent event is received
      * This is used for dynsacle
      */
-    internal fun updatePublishQuality(event: ChangePublishQualityEvent) {
-        if (publisher == null) {
-            return
+    internal fun updatePublishQuality(event: ChangePublishQualityEvent) = synchronized(this) {
+        val sender = publisher?.connection?.transceivers?.firstOrNull {
```

Comment:
I would log if this is null probably, something like "received `ChangePublishQualityEvent` but can't handle it because no video transceiver available". We have had quite a few hard to debug problems in the past due to random `return` in the code flow.